services/task-svc/internal/task/task.go

```diff
@@ -243,3 +452,55 @@ func (ServiceServer) UnassignTaskFromUser(ctx context.Context, req *pb.UnassignT
 
 	return &pb.UnassignTaskFromUserResponse{}, nil
 }
+
+func (ServiceServer) PublishTask(ctx context.Context, req *pb.PublishTaskRequest) (*pb.PublishTaskResponse, error) {
+	log := zlog.Ctx(ctx)
+	db := hwgorm.GetDB(ctx)
+
+	// TODO: Auth
+
+	id, err := uuid.Parse(req.Id)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	task := Task{ID: id}
+	updates := map[string]interface{}{"public": true}
+
+	if err := db.Model(&task).Updates(updates).Error; err != nil {
```

Comment:
Why not use `.Update("public", true)` here?

Reply:
Because I thought updates are always done with maps, but this is much more elegant